src/shape.cpp

```diff
@@ -346,12 +468,24 @@ const std::vector<shape>& shape::sub_shapes() const { return impl->m_shapes; }
 void migraphx_to_value(value& v, const shape& s)
 {
     value result;
-    result["type"]       = migraphx::to_value(s.type_string());
-    result["lens"]       = migraphx::to_value(s.lens());
-    result["strides"]    = migraphx::to_value(s.strides());
-    result["sub_shapes"] = migraphx::to_value(s.sub_shapes());
-    v                    = result;
+    result["type"] = migraphx::to_value(s.type_string());
+    if(s.dynamic())
+    {
+        result["dynamic"]      = migraphx::to_value(s.dynamic());
+        result["min_dyn_dims"] = migraphx::to_value(s.min_dyn_dims());
+        result["max_dyn_dims"] = migraphx::to_value(s.max_dyn_dims());
+        result["opt_dyn_dims"] = migraphx::to_value(s.opt_dyn_dims());
+        result["sub_shapes"]   = migraphx::to_value(s.sub_shapes());
+    }
+    else
+    {
+        result["lens"]       = migraphx::to_value(s.lens());
+        result["strides"]    = migraphx::to_value(s.strides());
+        result["sub_shapes"] = migraphx::to_value(s.sub_shapes());
+    }
```

Comment:
I dont think you need a branch just serialize it with:

```cpp
        result["lens"]       = migraphx::to_value(s.lens());
        result["strides"]    = migraphx::to_value(s.strides());
        result["sub_shapes"] = migraphx::to_value(s.sub_shapes());
        result["dynamic_dimensions"]       = migraphx::to_value(s.dyn_dims());
```

This should work after adding the `reflect` method.